src/main/java/teammates/logic/core/GoogleCloudLoggingService.java

```diff
@@ -170,13 +225,14 @@ private void createLogEntry(LogEntry entry) throws LogServiceException {
         return fsLogEntries;
     }
 
-    private List<LogEntry> getLogEntries(LogSearchParams s) throws LogServiceException {
-        List<LogEntry> logEntries = new ArrayList<>();
+    private Page<LogEntry> getLogEntries(LogSearchParams s, PageParams p) throws LogServiceException {
         LoggingOptions options = LoggingOptions.getDefaultInstance();
 
         List<String> logFilters = new ArrayList<>();
-        if (s.logName != null) {
-            logFilters.add("logName=\"projects/" + options.getProjectId() + "/logs/" + s.logName + "\"");
+        if (!s.logName.isEmpty()) {
+            String logNameFilter = s.logName.stream().reduce((subtotal, element) ->
+                    subtotal + "\" OR \"projects/" + options.getProjectId() + "/logs/" + element).get();
+            logFilters.add("logName=(\"projects/" + options.getProjectId() + "/logs/" + logNameFilter + "\")");
```

Comment:
`\"projects/" + options.getProjectId() + "/logs/"` is still being repeated. Try using `collect` and joining collectors instead of `reduce`.

Reply:
Ok will try using collect